src/lib/components/popup/Home.svelte

```diff
@@ -47,56 +47,82 @@
       listedCredentials = listedCredentials.map((cred) => ({
         ...cred,
         fields: cred.fields.filter(
-          (field) => field.fieldName !== "Domain" && field.fieldName !== "URL"
+          (field) => field.fieldName !== "Domain" && field.fieldName !== "URL",
         ),
       }));
     }
     const decyrptedResponse = await sendMessage(
       "decryptMeta",
-      listedCredentials
+      listedCredentials,
     );
     listedCredentials = decyrptedResponse.data;
     domainAssociatedCredentials = listedCredentials;
+  };
+  onMount(async () => {
+    query = await localStorage.getItem("query");
+    if (query.length >= 1) {
+      const searchFieldSResponse = await getSearchFields();
+      searchData = searchFieldSResponse.data;
+      listedCredentials = searchObjects(query, searchData);
+    } else {
+      await fetchCredentialsOfCurrentDomin();
+    }
     const user = await getUser();
     localStorage.setItem("user", JSON.stringify(user.data));
+    const storedCredentialId = localStorage.getItem("selectedCredentialId");
+    if (storedCredentialId != "") {
```

Comment:
what does this line do?

Reply:
while we are writing to local storage, if we write null as credential id, when we are reading the credential id the value will be "null" which is a string. hence we are checking for empty string rather than null string.

Reply:
if credentialId is not empty try dropdown and fetch the credential